app/controllers/users/invitations_controller.rb

```diff
@@ -33,5 +38,13 @@ def build_role_list
     def invite_params
       params.require(:user).permit(:email, :role, :name, :office_id)
     end
+
+    def user_for_authorisation
+      deleted_user_exists? || User.new(invite_params)
```

Comment:
You could cache the user `@user ||= ...` as it's called multiple times in the controller action.